utils/build-script-impl

```diff
@@ -2200,14 +2200,24 @@ for host in "${ALL_HOSTS[@]}"; do
                     )
                 fi
 
-                if [[ $(true_false "${SWIFT_ENABLE_EXPERIMENTAL_STRING_PROCESSING}") == "TRUE" && \
-                      -d "${EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR}" && \
+                if [[ -d "${EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR}" && \
                       -n "$(ls -A "${EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR}")" ]] ; then
                     cmake_options=(
                         "${cmake_options[@]}"
-                        -DSWIFT_ENABLE_EXPERIMENTAL_STRING_PROCESSING:BOOL=TRUE
                         -DEXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR="${EXPERIMENTAL_STRING_PROCESSING_SOURCE_DIR}"
                     )
+                    if [[ $(true_false "${SWIFT_ENABLE_EXPERIMENTAL_STRING_PROCESSING}") == "TRUE" ]] ; then
+                        cmake_options=(
+                            "${cmake_options[@]}"
+                            -DSWIFT_ENABLE_COMPILER_REGEX_PARSER:BOOL=TRUE
+                            -DSWIFT_ENABLE_EXPERIMENTAL_STRING_PROCESSING:BOOL=TRUE
+                        )
+                    elif [[ $(true_false "${SWIFT_ENABLE_COMPILER_REGEX_PARSER}") == "TRUE" ]] ; then
+                        cmake_options=(
+                            "${cmake_options[@]}"
+                            -DSWIFT_ENABLE_COMPILER_REGEX_PARSER:BOOL=TRUE
+                        )
```

Comment:
For this branch to work `swift-enable-compiler-regex-parser` would be to be added to `KNOWN_SETTINGS`, but possibly it could be better to remove this altogether and have the intended configurations to pass `SWIFT_ENABLE_COMPILER_REGEX_PARSER` as `extra-cmake-options`

Reply:
Actually, I don't know of anyone who explicitly _doesn't_ want to build the regex parser, except for places where the string processing repo is not checked out. So I don't think we need any build-script flag. Just an internal CMake flag would do.